src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    const modifiedArray = [];
+
+    for (const element of this) {
+      if (element === undefined || element === null) {
+        modifiedArray[modifiedArray.length] = '';
+      } else {
+        modifiedArray[modifiedArray.length] = element;
+      }
```

Comment:
delete else and replace === 2 !== in if and || 2 && and body of if - body of else

Reply:
done